src/main/java/io/jenkins/plugins/artifact_manager_s3/JCloudsArtifactManager.java

```diff
@@ -66,29 +62,26 @@
 import hudson.slaves.WorkspaceList;
 import hudson.util.DirScanner;
 import hudson.util.io.ArchiverFactory;
-import io.jenkins.plugins.artifact_manager_s3.JCloudsApiExtensionPoint.HttpMethod;
+import io.jenkins.plugins.artifact_manager_s3.BlobStoreProvider.HttpMethod;
 import jenkins.MasterToSlaveFileCallable;
 import jenkins.model.ArtifactManager;
 import jenkins.util.VirtualFile;
-import shaded.com.google.common.base.Supplier;
 
 /**
  * Artifact manager that stores files in a JClouds BlobStore using any of JClouds supported backends
  */
-class JCloudsArtifactManager extends ArtifactManager implements StashManager.StashAwareArtifactManager {
+final class JCloudsArtifactManager extends ArtifactManager implements StashManager.StashAwareArtifactManager {
 
     private static final Logger LOGGER = Logger.getLogger(JCloudsArtifactManager.class.getName());
 
-    private static String PROVIDER = System.getProperty("jclouds.provider", "aws-s3");
-
-    private static String BLOB_CONTAINER = System.getenv("S3_BUCKET");
-    private static String PREFIX = System.getenv("S3_DIR");
```

Comment:
Note that this would require trivial changes to #20.

Reply:
Fine with me. I am waiting for fixes for `runATH()` there